scripts/Interface/ui/drill/model/DrillModel.py

```diff
@@ -0,0 +1,714 @@
+            dict(str, any): key, value pairs of parameters. Value can be str,
+                            int, float, bool
+        """
+        params = dict()
+        params.update(self.settings)
+        params.update(self.samples[sample])
+        # override global params with custom ones
+        if "CustomOptions" in params:
+            params.update(params["CustomOptions"])
+            del params["CustomOptions"]
+        # remove empty params
+        for (k, v) in list(params.items()):
+            if v is None:
+                del params[k]
+        # add the output workspace param
+        if "OutputWorkspace" not in params:
+            params["OutputWorkspace"] = "sample_" + str(sample + 1)
+        return params
+
+    def process(self, elements):
+        """
+        Start samples processing.
+
+        Args:
+            elements (list(int)): list of sample indexes to be processed
+        """
+        tasks = list()
+        for e in elements:
+            if (e >= len(self.samples)) or (not self.samples[e]):
+                continue
+            kwargs = self.getProcessingParameters(e)
+            tasks.append(DrillTask(e, self.algorithm, **kwargs))
+        self.tasksPool.addProcesses(tasks)
+
+    def _onTaskStarted(self, ref):
+        """
+        Called each time a task starts.
+
+        Args:
+            ref (int): sample index
+        """
+        name = str(ref + 1)
+        logger.information("Starting of sample {0} processing"
+                           .format(name))
+        self.processStarted.emit(ref)
+
+    def _onTaskSuccess(self, ref):
+        """
+        Called when a task finished with success.
+
+        Args:
+            ref (int): sample index
+        """
+        name = str(ref + 1)
+        logger.information("Processing of sample {0} finished with sucess"
+                           .format(name))
+        self.processSuccess.emit(ref)
+
+    def _onTaskError(self, ref, msg):
+        """
+        Called when a processing fails. This method logs a message and fires the
+        corresponding signal.
+
+        Args:
+            ref (int): sample index
+            msg (str): error msg
+        """
+        name = str(ref + 1)
+        logger.error("Error while processing sample {0}: {1}"
+                     .format(name, msg))
+        self.processError.emit(ref)
+
+    def _onProcessingProgress(self, progress):
+        """
+        Called each time a processing updates its progress.
+
+        Args:
+            ref (int): sample index
+            progress (float): progress value (between 0.0 and 1.0)
+        """
+        self.progressUpdate.emit(progress)
+
+    def _onProcessingDone(self):
+        """
+        Called when the processing is done.
+        """
+        self.processingDone.emit()
+
+    def stopProcess(self):
+        """
+        Stop current processing.
+        """
+        self.tasksPool.abortProcessing()
+
+    def importRundexData(self, filename):
```

Comment:
Would it be easy to factor out the import/export logic to a separate class, e.g. DrillRundexIO ? In future this might be used also in autoreduction, so would be nice to decouple that from the GUI itself.

Reply:
As discussed, for now, import and export are really small:
* import is a JSON read and some checks (specific to the interface)
* export is a JSON write

We decided to keep them like that. They can be moved later.